Src/ILGPU/IR/Construction/Structures.cs

```diff
@@ -522,6 +522,49 @@ internal ValueReference FinishStructureBuilder<TBuilder>(ref TBuilder builder)
                 return instance.Seal();
             }
 
+            // Optimize common cases in which this set field operation fills a whole
+            // structure instance
+            if (objectValue is SetField &&
+                fieldSpan.Index + fieldSpan.Span == structureType.NumFields &&
+                structureType.NumFields < 32)
+            {
+                // Initialize our internal field-value lookup
+                var fieldValues = new Value[structureType.NumFields];
+                fieldValues[fieldSpan.Index] = value;
+
+                // Traverse the whole chain of all set-field operations
+                var parent = objectValue;
+                int traversalLength = structureType.NumFields - 1;
+                while (!(parent is NullValue) && traversalLength > 0)
+                {
+                    // Check whether we are still on the right track while traversing
+                    // set of (possibly unordered) set-field operations
+                    if (!(parent is SetField otherSetField) ||
+                        otherSetField.FieldSpan.HasSpan ||
+                        fieldValues[otherSetField.FieldSpan.Index] != null)
+                    {
+                        break;
+                    }
+
+                    // Register this set-field operation
+                    fieldValues[otherSetField.FieldSpan.Index] = otherSetField.Value;
+                    // Decrease the traversal length
+                    --traversalLength;
```

Comment:
ISSUE: This section also needs to deal with field spans greater than a single field.

Reply:
You are right 👍. After changing the if condition, we also need to update this code.

Reply:
After inspecting the code more closely, it turns out that this is not necessary, because we support single-element `SetField` operations only in the inner loop (see comment). The question is whether we should relax these restrictions.... I'm not sure if it's really worth the effort....

Reply:
yup, that's fine.
should we change `fieldSpan.Index + fieldSpan.Span` back to `fieldSpan.Index + 1`?